// paystack-sdk/dist/subaccounts/subaccount.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubAccount = void 0;
class SubAccount {
    constructor(http) {
        this.http = http;
    }
    create(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/subaccount', JSON.stringify(data));
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/subaccount', { params: Object.assign({}, queryParams) });
        });
    }
    fetch(id) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/subaccount/${id}`);
        });
    }
    update(id, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.put(`/subaccount/${id}`, JSON.stringify(data));
        });
    }
}
exports.SubAccount = SubAccount;
