// paystack-sdk/dist/apple/apple.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplePay = void 0;
class ApplePay {
    constructor(http) {
        this.http = http;
    }
    registerDomain(domainName) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/apple-pay/domain', JSON.stringify({ domainName }));
        });
    }
    listDomains() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/apple-pay');
        });
    }
    unregisterDomain(domainName) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.delete('/apple-pay', {
                params: { domainName },
            });
        });
    }
}
exports.ApplePay = ApplePay;
