// paystack-sdk/dist/transfer/control.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Control = void 0;
class Control {
    constructor(http) {
        this.http = http;
    }
    balance() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/balance');
        });
    }
    ledgerBalance() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/balance/ledger');
        });
    }
    resendOTP(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/transfer/resend_otp', JSON.stringify(data));
        });
    }
    disableOTP() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/transfer/disable_otp');
        });
    }
    finalizeDisableOTP(otp) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/transfer/disable_otp_finalize', JSON.stringify({ otp }));
        });
    }
    enableOTP() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/transfer/enable_otp');
        });
    }
}
exports.Control = Control;
