// paystack-sdk/dist/payment/payment.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentPage = void 0;
class PaymentPage {
    constructor(http) {
        this.http = http;
    }
    create(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/page', JSON.stringify(data));
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/page', { params: Object.assign({}, queryParams) });
        });
    }
    fetch(id) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/page/${id}`);
        });
    }
    update(id, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.put(`/page/${id}`, JSON.stringify(data));
        });
    }
    slugAvailable(slug) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/page/check_slug_availability/${slug}`);
        });
    }
    addProduct(id, products) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post(`/page/${id}/product`, JSON.stringify({ products }));
        });
    }
}
exports.PaymentPage = PaymentPage;
