// paystack-sdk/dist/product/product.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Product = void 0;
/**
 * @class Product
 * # Producs
 * The products API allows you create and manage inventories
 * on your integration
 */
class Product {
    constructor(http) {
        this.http = http;
    }
    create(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/product', JSON.stringify(data));
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/product', {
                params: Object.assign({}, queryParams),
            });
        });
    }
    fetch(id) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/product/${id}`);
        });
    }
    update(id, data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.put(`/product/${id}`, JSON.stringify(data));
        });
    }
}
exports.Product = Product;
