// paystack-sdk/dist/refund/refund.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Refund = void 0;
class Refund {
    constructor(http) {
        this.http = http;
    }
    /**
     * #### Create Refund
     * Initiate a refund on your integration
     */
    create(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/refund', JSON.stringify(data));
        });
    }
    /**
     * #### List Refunds
     * List refunds available on your integration
     */
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/refund', {
                params: Object.assign({}, queryParams),
            });
        });
    }
    /**
     * #### Fetch Refund
     * Get details of a refund on your integration
     */
    fetch(reference) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/refund/${reference}`);
        });
    }
}
exports.Refund = Refund;
