// paystack-sdk/dist/subscription/subscription.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Subscription = void 0;
class Subscription {
    constructor(http) {
        this.http = http;
    }
    create(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/subscription', JSON.stringify(data));
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/subscription', { params: Object.assign({}, queryParams) });
        });
    }
    fetch(idOrCode) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/subscription/${idOrCode}`);
        });
    }
    enable(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/subscription/enable', JSON.stringify(data));
        });
    }
    disable(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/subscription/disable', JSON.stringify(data));
        });
    }
    generateSubscriptionLink(code) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/subscription/${code}/manage/link`);
        });
    }
    sendUpdateSubscriptionLink(code) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post(`/subscription/${code}/manage/email`);
        });
    }
}
exports.Subscription = Subscription;
