// paystack-sdk/dist/dedicated/dedicated.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DedicatedAccount = void 0;
class DedicatedAccount {
    constructor(http) {
        this.http = http;
    }
    create(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/dedicated_account', JSON.stringify(data));
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/dedicated_account', {
                params: Object.assign({}, queryParams),
            });
        });
    }
    fetch(dedicatedAccountId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/dedicated_account/${dedicatedAccountId}`);
        });
    }
    deactivate(dedicatedAccountId) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.delete(`/dedicated_account/${dedicatedAccountId}`);
        });
    }
    splitTransaction(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/dedicated_account/split', JSON.stringify(data));
        });
    }
    removeSplit(accountNumber) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.delete('/dedicated_account/split', {
                data: { account_number: accountNumber },
            });
        });
    }
    providers() {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/dedicated_account/available_providers');
        });
    }
}
exports.DedicatedAccount = DedicatedAccount;
