// paystack-sdk/dist/bulkcharge/bulkcharge.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkCharge = void 0;
class BulkCharge {
    constructor(http) {
        this.http = http;
    }
    initiate(data) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.post('/bulkcharge', JSON.stringify(data));
        });
    }
    list(queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get('/bulkcharge', { params: Object.assign({}, queryParams) });
        });
    }
    fetchBulkCharge(id) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/bulkcharge/${id}`);
        });
    }
    fetchBatchChrges(id, queryParams) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/bulkcharge/${id}/charges`, {
                params: Object.assign({}, queryParams),
            });
        });
    }
    pause(batchCode) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/bulkcharge/pause/${batchCode}`);
        });
    }
    resume(batchCode) {
        return __awaiter(this, void 0, void 0, function* () {
            return yield this.http.get(`/bulkcharge/resume/${batchCode}`);
        });
    }
}
exports.BulkCharge = BulkCharge;
